constraint-solver/src/runtime_constant.rs

```diff
@@ -0,0 +1,97 @@
+use std::ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub};
+
+use num_traits::{One, Zero};
+use powdr_number::FieldElement;
+
+use crate::{
+    quadratic_symbolic_expression::QuadraticSymbolicExpressionImpl,
+    range_constraint::RangeConstraint,
+};
+
+/// Represents a run-time constant in the constraint solver.
+/// Any T: FieldElement can represent a run-time constant (which is also a compile-time constant),
+/// but the trait lets us represent run-time constants symbolically as well.
+pub trait RuntimeConstant<V>:
+    Sized
+    + Neg<Output = Self>
+    + Clone
+    + From<Self::FieldType>
+    + Add<Output = Self>
+    + AddAssign<Self>
+    + Sub<Output = Self>
+    + Mul<Output = Self>
+    + MulAssign<Self>
+    + PartialEq
+    + Eq
+    + Zero
+    + One
+    + TryInto<QuadraticSymbolicExpressionImpl<Self, V>>
```

Comment:
Why is this needed? Should RuntimeCosntant know about QSE?

Reply:
Hmm, this is currently where we put this:
https://github.com/powdr-labs/powdr/blob/62046ceb2cdcaba3768e7bb0b0769bd3b98349c6/constraint-solver/src/symbolic_expression.rs#L465-L473

And for `FieldElement`, it is already implemented because QSE implements `From<T: FieldElement>`.

I think it's fine if `RuntimeConstant` knows QSE, because it is only used in conjunction with that?